Philips.CodeAnalysis.MaintainabilityAnalyzers/CopyrightPresentAnalyzer.cs

```diff
@@ -76,5 +81,22 @@ private static void CreateDiagnostic(SyntaxNodeAnalysisContext context, Location
 			Diagnostic diagnostic = Diagnostic.Create(Rule, location);
 			context.ReportDiagnostic(diagnostic);
 		}
+
+		private static bool CheckCopyrightStatement(SyntaxNodeAnalysisContext context, SyntaxTrivia trivia) {
+			var comment = trivia.ToFullString();
+			// Check the copyright mar itself
+			var hasCopyright = comment.Contains("©") || comment.Contains("Copyright");
+			
+			// Check the year
+			var yearRegex = new Regex(@"\d\d\d\d");
```

Comment:
A Regex here feels like a relatively expensive operation to run for every compilation unit, but I don't have data to back up this concern of a performance hit.

Reply:
Made the RegEx a static variable. This at least save the RegEx compile time.

Had to put a lock around it, as a save guard during concurrent execution.